com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/cf/v3_1/ApplicationsCloudModelBuilder.java

```diff
@@ -0,0 +1,146 @@
+        UserMessageLogger userMessageLogger) {
+        super(deploymentDescriptor, configuration, deployedMta, systemParameters, xsPlaceholderResolver, deployId, userMessageLogger);
+    }
+    
+    @Override
+    protected HandlerFactory getHandlerFactory() {
+        return new HandlerFactory(MTA_MAJOR_VERSION, MTA_MINOR_VERSION);
+    }
+    
+    @Override
+    protected CloudApplicationExtended getApplication(com.sap.cloud.lm.sl.mta.model.v1_0.Module module) throws SLException {
+        DeployedMtaModule deployedModule = findDeployedModule(deployedMta, module);
+        List<Map<String, Object>> parametersList = parametersChainBuilder.buildModuleChain(module.getName());
+        warnAboutUnsupportedParameters(parametersList);
+        Staging staging = parseParameters(parametersList, new StagingParametersParser());
+        int diskQuota = parseParameters(parametersList, new MemoryParametersParser(SupportedParameters.DISK_QUOTA, "0"));
+        int memory = parseParameters(parametersList, new MemoryParametersParser(SupportedParameters.MEMORY, "0"));
+        int instances = (Integer) getPropertyValue(parametersList, SupportedParameters.INSTANCES, 0);
+        List<String> uris = urisCloudModelBuilder.getApplicationUris(module, parametersList);
+        List<String> idleUris = urisCloudModelBuilder.getIdleApplicationUris(module, parametersList);
+        List<String> resolvedUris = xsPlaceholderResolver.resolve(uris);
+        List<String> resolvedIdleUris = xsPlaceholderResolver.resolve(idleUris);
+        List<String> customUris = new UrisClassifier(xsPlaceholderResolver).getCustomUris(deployedModule);
+        List<String> fullResolvedUris = ListUtil.merge(resolvedUris, customUris);
+        List<String> services = getActiveApplicationServices(module);
+        List<ServiceKeyToInject> serviceKeys = getServicesKeysToInject(module);
+        Map<Object, Object> env = applicationEnvCloudModelBuilder.build(module, uris, getApplicationServices(module),
+            getSharedApplicationServices(module), module.getProperties(), ((Module) module).getParameters());
+        List<CloudTask> tasks = getTasks(parametersList);
+        Map<String, Map<String, Object>> bindingParameters = getBindingParameters((Module) module);
+        List<ApplicationPort> applicationPorts = getApplicationPorts((com.sap.cloud.lm.sl.mta.model.v2_0.Module) module, parametersList);
+        List<String> applicationDomains = getApplicationDomains((com.sap.cloud.lm.sl.mta.model.v2_0.Module) module, parametersList);
+        return createCloudApplication(getApplicationName(module), module.getName(), staging, diskQuota, memory, instances, fullResolvedUris,
+            resolvedIdleUris, services, serviceKeys, env, bindingParameters, tasks, applicationPorts, applicationDomains);
+    }
+    
+    protected List<String> getActiveApplicationServices(com.sap.cloud.lm.sl.mta.model.v1_0.Module module) {
```

Comment:
suggestion: since all the service filter methods include filtering for active only maybe this should remain as getAllApplicationServices to be less confusing?